Test gym changes to manually test client disconnect (#346)
spatialos/UnrealGDKTestGyms: USER_MANUAL.md

 1. Select `Play` on the Unreal toolbar.
 1. Watch as the cubes turn green in under 8 seconds.

##### Player disconnect gym
* Demonstrates that:
  * Players are cleaned up correctly when they disconnect.
* NOTE: This gym can only be run manually. (Note: this can be automated once the test frameworks supports client disconnects https://improbableio.atlassian.net/browse/UNR-529)
* Pre-test steps:
  * Navigate to: Project Settings-> Project - Maps & Modes -> Game Instance and set the Game Instance Class to `PlayerDisconnectGameInstance`
  * In the Unreal Editor's Content Browser, locate `Content/Maps/PlayerDisconnectGym` and double click to open it.

> * Client disconnect test:

Reviewer: ```suggestion
* Test for a client disconnecting by returning to the main menu:
```

  * In the Unreal Editor's Toolbar, on the play dropdown menu select two players and then click `Play` button

>   * Verify in Inspector, two client workers, two player controllers (PlayerDisconnectController) and two player characters.

Reviewer: ```suggestion
  * Verify in Inspector that two client worker entities, two player controller entities (PlayerDisconnectController) and two player character entities exist.
```

Reviewer: Similarly for the identical lines below.

>   * Press 'M' key for one client, to travel this player to the empty map.

Reviewer: ```suggestion
  * Press 'M' in one of the clients, to make that client leave the deployment by traveling to the empty map.
```

>   * Verify in the Inspector, one client worker, one player controller and one player character.

Reviewer: ```suggestion
  * Verify in the Inspector that only one client worker entity, one player controller entity and one player character entity exist.
```

> * Server disconnects all clients test:

Reviewer: ```suggestion
* Test for the server disconnecting all clients:
```

>   * In the Unreal Editor's Toolbar, on the play dropdown menu select two players and then click `Play` button

Reviewer: ```suggestion
  * Shut down the previous deployment
  * In the Unreal Editor's Toolbar, on the play dropdown menu select two players and then click `Play` button
```

  * Verify in Inspector, two client workers, two player controllers (PlayerDisconnectController) and two player characters.
  * Open a command window and go to the spatial directory of the GDKTestGyms, use the following command: curl -X PUT -d "Yes" localhost:5006/worker_flag/workers/UnrealWorker/flags/PrepareShutdown
  * Verify in the Inspector, zero client workers, zero player controllers and zero player characters.

> * Close client window test:

Reviewer: ```suggestion
* Test for a client disconnecting by closing their window:
```
